Build failure on Linux (musl)

Not entirely sure why the build is failing, since Linux CI is working. The build occurs in a git checkout, not from a release tarball.

I'm thinking that the build failure is musl specific, at least in regards to:

```
error: cannot assign to variable 'stdin' with const-qualified type 'FILE *const' (aka '_IO_FILE *const')
```

## Build Env:
- SBCL 2.3.1
- clang version 15.0.7
- musl
- Linux 5.10.135

## Build Script:

```shell
#!/bin/sh -e

export CC=clang
export CXX=clang++

sbcl --script koga \
    --reproducible-build \
    --broken-stdlib \
    --package-path="$1" \
    --bin-path=/usr/bin \
    --share-path=/usr/share/clasp/ \
    --lib-path=/usr/lib/clasp

ninja -C build
ninja -C build install
```

## Build Log:

```
[376/543] Compiling ../src/gctools/exposeFunctions1.cc
ninja: job failed: /usr/bin/clang++ -Iboehmprecise -Iboehmprecise/generated -O3 -g -fPIC -ffile-prefix-map=boehmprecise/generated=/usr/share/clasp/generated -ffile-prefix-map=boehmprecise/lib=/usr/lib/clasp -Wno-macro-redefined -Wno-deprecated-declarations -Wno-deprecated-register -Wno-expansion-to-defined -Wno-return-type-c-linkage -Wno-invalid-offsetof -Wno-#pragma-messages -Wno-inconsistent-missing-override -Wno-error=c++11-narrowing -Wno-c++11-narrowing -Wno-deprecated-enum-enum-conversion -Wno-deprecated-anon-enum-enum-conversion -std=c++17 -Wno-c++20-extensions -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -fno-stack-protector -stdlib=libstdc++ -I/usr/include -ffile-prefix-map=..=/usr/share/clasp -I.. -I../src/bdwgc -I../src/bdwgc/include -I../src/libatomic_ops/src -I../include -I../include/clasp/main -c -MD -MF boehmprecise/src/gctools/mpsGarbageCollection.o.d -oboehmprecise/src/gctools/mpsGarbageCollection.o ../src/gctools/mpsGarbageCollection.cc
In file included from ../src/gctools/mpsGarbageCollection.cc:67:
../include/clasp/core/lispStream.h:60:48: error: unknown type name 'mode_t'; did you mean 'Code_S'?
int safe_open(const char *filename, int flags, clasp_mode_t mode);
                                               ^~~~~~~~~~~~
                                               Code_S
../include/clasp/core/lispStream.h:57:22: note: expanded from macro 'clasp_mode_t'
#define clasp_mode_t mode_t
                     ^
../include/clasp/gctools/pointer_tagging.h:65:9: note: 'Code_S' declared here
  class Code_S;
        ^
1 error generated.
ninja: job failed: /usr/bin/clang++ -Iboehmprecise -Iboehmprecise/generated -O3 -g -fPIC -ffile-prefix-map=boehmprecise/generated=/usr/share/clasp/generated -ffile-prefix-map=boehmprecise/lib=/usr/lib/clasp -Wno-macro-redefined -Wno-deprecated-declarations -Wno-deprecated-register -Wno-expansion-to-defined -Wno-return-type-c-linkage -Wno-invalid-offsetof -Wno-#pragma-messages -Wno-inconsistent-missing-override -Wno-error=c++11-narrowing -Wno-c++11-narrowing -Wno-deprecated-enum-enum-conversion -Wno-deprecated-anon-enum-enum-conversion -std=c++17 -Wno-c++20-extensions -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -fno-stack-protector -stdlib=libstdc++ -I/usr/include -ffile-prefix-map=..=/usr/share/clasp -I.. -I../src/bdwgc -I../src/bdwgc/include -I../src/libatomic_ops/src -I../include -I../include/clasp/main -c -MD -MF boehmprecise/src/gctools/snapshotSaveLoad.o.d -oboehmprecise/src/gctools/snapshotSaveLoad.o ../src/gctools/snapshotSaveLoad.cc
../src/gctools/snapshotSaveLoad.cc:2582:11: error: cannot assign to variable 'stdin' with const-qualified type 'FILE *const' (aka '_IO_FILE *const')
    stdin = freopen(NULL,"r",stdin);
    ~~~~~ ^
/usr/include/stdio.h:62:20: note: variable 'stdin' declared const here
extern FILE *const stdin;
~~~~~~~~~~~~~~~~~~~^~~~~
1 error generated.
ninja: job failed: /usr/bin/clang++ -Iboehmprecise -Iboehmprecise/generated -O3 -g -fPIC -ffile-prefix-map=boehmprecise/generated=/usr/share/clasp/generated -ffile-prefix-map=boehmprecise/lib=/usr/lib/clasp -Wno-macro-redefined -Wno-deprecated-declarations -Wno-deprecated-register -Wno-expansion-to-defined -Wno-return-type-c-linkage -Wno-invalid-offsetof -Wno-#pragma-messages -Wno-inconsistent-missing-override -Wno-error=c++11-narrowing -Wno-c++11-narrowing -Wno-deprecated-enum-enum-conversion -Wno-deprecated-anon-enum-enum-conversion -std=c++17 -Wno-c++20-extensions -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -fno-stack-protector -stdlib=libstdc++ -I/usr/include -ffile-prefix-map=..=/usr/share/clasp -I.. -I../src/bdwgc -I../src/bdwgc/include -I../src/libatomic_ops/src -I../include -I../include/clasp/main -c -MD -MF boehmprecise/src/gctools/gcFunctions.o.d -oboehmprecise/src/gctools/gcFunctions.o ../src/gctools/gcFunctions.cc
In file included from ../src/gctools/gcFunctions.cc:25:
In file included from ../include/clasp/core/bformat.h:31:
../include/clasp/core/lispStream.h:60:48: error: unknown type name 'mode_t'; did you mean 'Code_S'?
int safe_open(const char *filename, int flags, clasp_mode_t mode);
                                               ^~~~~~~~~~~~
                                               Code_S
../include/clasp/core/lispStream.h:57:22: note: expanded from macro 'clasp_mode_t'
#define clasp_mode_t mode_t
                     ^
../include/clasp/gctools/pointer_tagging.h:65:9: note: 'Code_S' declared here
  class Code_S;
        ^
1 error generated.
ninja: job failed: /usr/bin/clang++ -Iboehmprecise -Iboehmprecise/generated -O3 -g -fPIC -ffile-prefix-map=boehmprecise/generated=/usr/share/clasp/generated -ffile-prefix-map=boehmprecise/lib=/usr/lib/clasp -Wno-macro-redefined -Wno-deprecated-declarations -Wno-deprecated-register -Wno-expansion-to-defined -Wno-return-type-c-linkage -Wno-invalid-offsetof -Wno-#pragma-messages -Wno-inconsistent-missing-override -Wno-error=c++11-narrowing -Wno-c++11-narrowing -Wno-deprecated-enum-enum-conversion -Wno-deprecated-anon-enum-enum-conversion -std=c++17 -Wno-c++20-extensions -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -fno-stack-protector -stdlib=libstdc++ -I/usr/include -ffile-prefix-map=..=/usr/share/clasp -I.. -I../src/bdwgc -I../src/bdwgc/include -I../src/libatomic_ops/src -I../include -I../include/clasp/main -c -MD -MF boehmprecise/src/gctools/boehmGarbageCollection.o.d -oboehmprecise/src/gctools/boehmGarbageCollection.o ../src/gctools/boehmGarbageCollection.cc
In file included from ../src/gctools/boehmGarbageCollection.cc:31:
../include/clasp/core/lispStream.h:60:48: error: unknown type name 'mode_t'; did you mean 'Code_S'?
int safe_open(const char *filename, int flags, clasp_mode_t mode);
                                               ^~~~~~~~~~~~
                                               Code_S
../include/clasp/core/lispStream.h:57:22: note: expanded from macro 'clasp_mode_t'
#define clasp_mode_t mode_t
                     ^
../include/clasp/gctools/pointer_tagging.h:65:9: note: 'Code_S' declared here
  class Code_S;
        ^
1 error generated.
ninja: subcommands failed
```

Full Log: https://0x0.st/Hsew.okHoiK

Seems like the build gets a little farther before erroring out again, now it's upset about the use of `NETDB_INTERNAL`:

```
[394/543] Compiling ../src/llvmo/clbindLlvmExpose.cc
ninja: job failed: /usr/bin/clang++ -Iboehmprecise -Iboehmprecise/generated -O3 -g -fPIC -ffile-prefix-map=boehmprecise/generated=/usr/share/clasp/generated -ffile-prefix-map=boehmprecise/lib=/usr/lib/clasp -Wno-macro-redefined -Wno-deprecated-declarations -Wno-deprecated-register -Wno-expansion-to-defined -Wno-return-type-c-linkage -Wno-invalid-offsetof -Wno-#pragma-messages -Wno-inconsistent-missing-override -Wno-error=c++11-narrowing -Wno-c++11-narrowing -Wno-deprecated-enum-enum-conversion -Wno-deprecated-anon-enum-enum-conversion -std=c++20 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -fno-stack-protector -stdlib=libstdc++ -I/usr/include -ffile-prefix-map=..=/usr/share/clasp -I.. -I../src/bdwgc -I../src/bdwgc/include -I../src/libatomic_ops/src -I../include -I../include/clasp/main -c -MD -MF boehmprecise/src/sockets/sockets.o.d -oboehmprecise/src/sockets/sockets.o ../src/sockets/sockets.cc
../src/sockets/sockets.cc:930:84: error: use of undeclared identifier 'NETDB_INTERNAL'
  _sym__PLUS_NETDB_INTERNAL_PLUS_->defconstant(core::Integer_O::create((gc::Fixnum)NETDB_INTERNAL));
                                                                                   ^
../src/sockets/sockets.cc:932:83: error: use of undeclared identifier 'NETDB_SUCCESS'
  _sym__PLUS_NETDB_SUCCESS_PLUS_->defconstant(core::Integer_O::create((gc::Fixnum)NETDB_SUCCESS));
                                                                                  ^
2 errors generated.
ninja: subcommand failed
```

This also looks like a musl error.

```
[411/543] Compiling ../src/asttooling/asttoolingPackage.cc
ninja: job failed: /usr/bin/clang++ -Iboehmprecise -Iboehmprecise/generated -O3 -g -fPIC -ffile-prefix-map=boehmprecise/generated=/usr/share/clasp/generated -ffile-prefix-map=boehmprecise/lib=/usr/lib/clasp -Wno-macro-redefined -Wno-deprecated-declarations -Wno-deprecated-register -Wno-expansion-to-defined -Wno-return-type-c-linkage -Wno-invalid-offsetof -Wno-#pragma-messages -Wno-inconsistent-missing-override -Wno-error=c++11-narrowing -Wno-c++11-narrowing -Wno-deprecated-enum-enum-conversion -Wno-deprecated-anon-enum-enum-conversion -std=c++20 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -fno-stack-protector -stdlib=libstdc++ -I/usr/include -ffile-prefix-map=..=/usr/share/clasp -I.. -I../src/bdwgc -I../src/bdwgc/include -I../src/libatomic_ops/src -I../include -I../include/clasp/main -c -MD -MF boehmprecise/src/llvmo/link_intrinsics.o.d -oboehmprecise/src/llvmo/link_intrinsics.o ../src/llvmo/link_intrinsics.cc
../src/llvmo/link_intrinsics.cc:260:3: error: use of undeclared identifier 'va_start'
  va_start(va, len);
  ^
../src/llvmo/link_intrinsics.cc:263:40: error: 'T_O' does not refer to a value
    cons->rplaca(core::T_sp(va_arg(va, T_O*)));
                                       ^
../include/clasp/core/object.h:400:9: note: declared here
  class T_O : public RootClass {
        ^
../src/llvmo/link_intrinsics.cc:263:44: error: expected expression
    cons->rplaca(core::T_sp(va_arg(va, T_O*)));
                                           ^
../src/llvmo/link_intrinsics.cc:657:3: error: use of undeclared identifier 'va_start'
  va_start(args, nargs);
  ^
../src/llvmo/link_intrinsics.cc:660:56: error: 'T_O' does not refer to a value
    T_O* tagged_obj = ENSURE_VALID_OBJECT(va_arg(args, T_O*));
                                                       ^
../include/clasp/core/object.h:400:9: note: declared here
  class T_O : public RootClass {
        ^
../src/llvmo/link_intrinsics.cc:660:60: error: expected expression
    T_O* tagged_obj = ENSURE_VALID_OBJECT(va_arg(args, T_O*));
                                                           ^
../src/llvmo/link_intrinsics.cc:663:3: error: use of undeclared identifier 'va_end'
  va_end(args);
  ^
../src/llvmo/link_intrinsics.cc:1150:3: error: use of undeclared identifier 'va_start'
  va_start(argp, numCells);
  ^
../src/llvmo/link_intrinsics.cc:1154:52: error: expected '(' for function-style cast or type construction
    p = ENSURE_VALID_OBJECT(va_arg(argp, core::T_O *));
                                         ~~~~~~~~~ ^
../include/clasp/gctools/memoryManagement.h:1223:32: note: expanded from macro 'ENSURE_VALID_OBJECT'
#define ENSURE_VALID_OBJECT(x) x
                               ^
../src/llvmo/link_intrinsics.cc:1154:53: error: expected expression
    p = ENSURE_VALID_OBJECT(va_arg(argp, core::T_O *));
                                                    ^
../src/llvmo/link_intrinsics.cc:1158:3: error: use of undeclared identifier 'va_end'
  va_end(argp);
  ^
11 errors generated.
ninja: subcommand failed
```

--- a/src/gctools/snapshotSaveLoad.cc
+++ b/src/gctools/snapshotSaveLoad.cc
@@ -2579,7 +2579,7 @@ void* snapshot_save_impl(void* data) {
     printf("%s:%d PID = %d  Paused at exit - press enter to continue: \n", __FILE__, __LINE__, getpid() );
     fflush(stdout);
     fseek(stdin,0,SEEK_END);
-    stdin = freopen(NULL,"r",stdin);
+    freopen(NULL,"r",stdin);
     int c;
     while ( (c = getchar()) != '\n' && c != EOF ) { };
   }

